[FEAT] 따봉도치 디테일뷰 연결

ubeeni opened the pull request:
## 작업한 내용
- 따봉도치 디테일뷰 연결
  - 오늘의 따봉도치
  - 따봉도치 둘러보기
  - 행운별 모아보기
  - 마이페이지/나의 따봉도치

## 📸 스크린샷
https://github.com/dnd-side-project/dnd-10th-9-iOS/assets/69234788/e71b7a71-1a33-4e77-b42b-9265c3111975

https://github.com/dnd-side-project/dnd-10th-9-iOS/assets/69234788/f596750b-d5ce-45b8-9948-649e06f40ba7

## 관련 이슈
- Resolved: #64

ubeeni committed:
chore: APIConstants.pagingDefaultValue 수정 #64

ubeeni committed:
chore: APIConstants.pagingDefaultValue 수정 #64

ubeeni committed:
feat: 오늘의 따봉도치 디테일뷰 연결

- 이미지 클릭시 이동

ubeeni committed:
feat: 오늘의 따봉도치 디테일뷰 연결

- 이미지 클릭시 이동

ubeeni committed:
feat: 따봉도치 둘러보기 디테일뷰 연결 #64

ubeeni committed:
feat: 따봉도치 둘러보기 디테일뷰 연결 #64

ubeeni committed:
chore: 따봉도치 둘러보기(recentCards) 카드 이미지, 컬러 수정 #64

ubeeni committed:
chore: 따봉도치 둘러보기(recentCards) 카드 이미지, 컬러 수정 #64

ubeeni committed:
feat: 행운별 모아보기 디테일뷰 연결 #64

ubeeni committed:
feat: 행운별 모아보기 디테일뷰 연결 #64

ubeeni committed:
feat: 나의 따봉도치 디테일뷰 연결 #64

ubeeni committed:
feat: 나의 따봉도치 디테일뷰 연결 #64

dev-jungbin commented on DOTCHI/DOTCHI/Sources/Screens/Home/CollectionView.swift:
```diff
@@ -81,7 +81,7 @@ struct CollectionView: View {
         }
         .onAppear() {
-            themeViewModel.fetchTheme(themeId: themeId, cardSortType: selectedButton, lastCardId: 999999999, lastCardCommentCount: 999999999)
+            themeViewModel.fetchTheme(themeId: themeId, cardSortType: selectedButton, lastCardId: APIConstants.pagingDefaultValue, lastCardCommentCount: APIConstants.pagingDefaultValue)
```
👍

dev-jungbin commented on DOTCHI/DOTCHI/Sources/Screens/Home/HomeView.swift:
```diff
@@ -7,6 +7,17 @@
 
 import SwiftUI
 
+struct DotchiDetailViewControllerWrapper: UIViewControllerRepresentable {
```
이미 DotchiDetailViewControllerRepresentable 을 만들어 뒀어가지구.. 이거 사용하시는 게 어떨까여..?!?!?
![스크린샷 2024-02-23 오후 11 51 53](https://github.com/dnd-side-project/dnd-10th-9-iOS/assets/43312096/4c4512cc-a539-4ac2-85f4-3f93d2aab2c3)

ubeeni replied:
`DotchiDetailViewControllerRepresentable(isPresented: $isDetailPresented, cardId: today.cardId)` 로 해서 써봤는데 백버튼 작동이 안 되는데 이유가 있을까요..?

dev-jungbin replied:
오잉?? 안에 코드 똑같은데 isPresented가 문제일까여,,
isPresented 이 파라미터는 혹시 몰라서 넣어둔거라
프로퍼티, 파라미터 모두 지우셔도 됩니당!!

ubeeni replied:
이게 DotchiDetailViewController 자체에서 내비게이션을 처리하고 있어서 DotchiDetailViewController의 인스턴스를 직접 반환해줘야 제대로 작동을 한다고 하네요...? 혹시 `DotchiDetailViewControllerRepresentable` 파일 코드 자체를 수정해도 될까요? 이 코드를 다른 부분에서 쓸 것 같으면 기존 방법대로 따로 만들어서 사용하는게 좋을 것 같아요..!

dev-jungbin replied:
@ubeeni DotchiDetailViewControllerRepresentable 로 검색해보시면 지금 다른 파일에서 안 쓰고 있어요!!! 코드 자체 수정하셔도 됩니당 ~!

ubeeni replied:
넵 그럼 코드 자체를 그냥 수정해서 사용하겠습니다!

dev-jungbin reviewed (changes requested):
리뷰 확인해주세여 !!!!
근데 지금 navigation link로 넘어가는데, , , 

지금 카드뷰 레이아웃 이슈만 잡고 
탭바 문제 때문에 DetailViewcontroller로 넘어갈 때 X버튼 달린 present modal로 변경하려고 합니다 ! 혹시 이때는 어케 수정해야...할까요?? (제가할게요..알려주시면 . . .)

dev-jungbin committed:
chore: dotchi card front view 이미지 너무 확대되어 보이던 문제 수정 #72

dev-jungbin committed:
chore: 따봉도치 만들기 뷰에서도 photo image view layout 적용 #72

ubeeni commented:
```
.fullScreenCover(isPresented: $isProfileEditViewPresented, content: {
                        ProfileEditView()
                            .transition(.move(edge: .trailing))
                    })
```
마이페이지에서 수정하기 뷰로 넘어갈 때 썼던 코드입니다. 이렇게 하면 되지 않을까요..?

dev-jungbin commented:
> ```
> .fullScreenCover(isPresented: $isProfileEditViewPresented, content: {
>                         ProfileEditView()
>                             .transition(.move(edge: .trailing))
>                     })
> ```
> 
> 마이페이지에서 수정하기 뷰로 넘어갈 때 썼던 코드입니다. 이렇게 하면 되지 않을까요..?

감사합니당 !! 지금 이슈 작업하고 요거 적용해볼게여 ..!!

dev-jungbin committed:
chore: detail View에서 luckyType 적용 안 하고 있던 문제 수정 #76

dev-jungbin committed:
chore: BaseViewController에 setCloseButtonAction 메서드 구현 #78

dev-jungbin committed:
chore: DotchiNavigationView에 .closeMore 케이스 구현 #78

dev-jungbin committed:
chore: 따봉도치 만들기 뷰 close button action 주는 방식 수정 #78

dev-jungbin committed:
chore: DotchiDetail 뷰에서 네비게이션 뷰 .closeMore로 변경 #78

dev-jungbin committed:
chore: Browse -> DotchiDetail 전환 방식 present로 변경 #78

dev-jungbin committed:
[CHORE] card dotchiImageView layout 수정 (#75)

## 작업한 내용
- 도치 이미지가 너무 확대되어 보여서, imageView 위치/크기 수정하였습니다!

## 🎶 PR Point
<!-- 피드백을 받고 싶은 부분, 공유하고 싶은 부분, 작업 과정, 이유를 적어주세요. -->
- 나중에 유빈님 뷰에도 수정하실 때 참고하시라고 constraints 드려요!!
![스크린샷 2024-02-24 오전 12 06
03](https://github.com/dnd-side-project/dnd-10th-9-iOS/assets/43312096/882517b2-fb39-4981-ad30-86d8c41f77b9)


## 📸 스크린샷
|before|after|
|---|---|

|![93BC28B6-B578-417E-A52C-B867411D10C7](https://github.com/dnd-side-project/dnd-10th-9-iOS/assets/43312096/80fad0c7-8660-48e7-ba78-2d30f3f57be7)|![3AE4A5D3-CC2B-4D6A-9A00-8ECF88EF16E6](https://github.com/dnd-side-project/dnd-10th-9-iOS/assets/43312096/2d040159-d194-4008-a92a-6dbaeebeb7a6)|

### Hierarchy
|before|after|
|---|---|
|![스크린샷 2024-02-23 오후 11 50
25](https://github.com/dnd-side-project/dnd-10th-9-iOS/assets/43312096/8ffb82b6-9daa-462f-a605-d0499639ce38)|![스크린샷
2024-02-24 오전 12 05
50](https://github.com/dnd-side-project/dnd-10th-9-iOS/assets/43312096/e2a95e0d-cfa6-4a22-b3e7-bd371f4ba4ea)|





## 관련 이슈
- Resolved: #72

dev-jungbin committed:
[FIX] 따봉도치 디테일 - 댓글의 luckyType 적용 안 되는 문제 수정 (#77)

## 작업한 내용
- 따봉도치 디테일 - 댓글의 luckyType 적용 안 되는 문제 수정
- 유빈님 감사합니다!!!!!!!! 🙇‍♀️

## 📸 스크린샷

![image](https://github.com/dnd-side-project/dnd-10th-9-iOS/assets/43312096/743e330c-ebb5-4d1a-91c9-ffff1670daf7)


## 관련 이슈
- Resolved: #76

dev-jungbin committed:
[CHORE] 따봉도치 디테일 뷰 present modal로 열리도록 변경 (#79)

## 작업한 내용
- 따봉도치 디테일 뷰 present modal로 열리도록 변경
- navigation view .closeMore 케이스 추가

## 📸 스크린샷


https://github.com/dnd-side-project/dnd-10th-9-iOS/assets/43312096/84c4a692-0440-4c7c-8890-7011d15d782c



## 관련 이슈
- Resolved: #78

ubeeni committed:
chore: 디테일뷰 연결 방식 변경 #64

ubeeni committed:
Merge branch 'feat/#64-Connect' of https://github.com/dnd-side-project/dnd-10th-9-iOS into feat/#64-Connect

dev-jungbin reviewed (approved):
넘넘넘 수고하쎳습니다~~~ ><
